plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+
+If you encounter any issues while using the Backstage version of the SDK, please don't hesitate to open an issue. We would be happy to get any issues fixed, but expect some growing pains as we collect user feedback.
+
+## Welcome to the Coder SDK!
+
+The Coder SDK for Backstage makes it easy for Backstage admins to bring the entire Coder API into Spotify's Backstage platform. While the Coder Backstage plugin does ship with a collection of ready-made components, those can't meet every user's needs, and so, why not give you access to the full set of building blocks, so you can build a solution tailored to your specific use case?
+
+### Table of contents
+
+This guide covers the following:
+
+- Accessing the SDK from your own custom React components
+  - Accessing the SDK via `useCoderSdk`
+  - Accessing the SDK via `useCoderQuery`
+- Authenticating
+  - The fallback auth UI
+- Performing queries
+  - Recommendations for caching data
+  - How the Coder plugin caches data
+  - The `useCoderQuery` custom hook
+  - Cautions against other common UI caching strategies
+- Performing mutations
+
+### Before you begin
+
+This guide assumes that you have already added the `CoderProvider` component to your Backstage deployment. If you have not, [please see the main README](../../README.md#setup) for instructions on getting that set up.
+
+## Quick-start
+
+### The main hooks to use when working with the SDK
+
+There are about six main hooks that you want to consider when using the Coder SDK.These can be broken down into two main categories:
+
+#### Primitive hooks
+
+- `useCoderSdk` - Gives you the Coder SDK instance
+- `useCoderAuth` - Exposes properties and methods for examining/updating Coder auth state
+- `useQuery` (via Tanstack Query library) - Declarative querying
+- `useMutation` (via Tanstack Query library) - Imperative mutations
+- `useQueryClient` (via Tanstack Query library) - Lets you coordinate queries and mutations within the same query cache
+
+#### Convenience hooks
+
+- `useCoderQuery` (wrapper over `useQuery`)
+
+## Accessing the SDK from your own custom React components
+
+There are two main ways of accessing the Coder SDK:
+
+- `useCoderSdk`
+- `useCoderQuery`
+
+### Accessing the SDK through `useCoderSdk`
+
+`useCoderSdk` is a lower-level "primitive" for accessing the Coder SDK. The SDK exposes a mix of different REST API methods for interacting with your Coder deployment's resources.
+
+Calling the hook will give you an object with all available API methods. As these methods are all async, **none** of them are suitable for use in render logic. They must be called from within effects or event handlers.
+
+```tsx
+// Illustrative example - these patterns are a very bad idea in production!
```

Comment:
I get why you put this warning here but combined with the async reminder in the paragraph above, it took me a second to understand you were exemplifying usage (rather than including a 'don't do this' snippet). Maybe we could consider rewording slightly?